Update IAR Armv8M changes - Work in progress

deepikabhavnani opened the pull request:
### Description

Update IAR Armv8M changes ~~and IAR version~~ Not updating IAR version, will be done as part of feature branch merge. 

feature-iar8 needs to be tested with IAR 8.32 version - Update needed in CI

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

mmahadevan108 committed:
Add nanostack support for KW41Z

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

kegilbert committed:
Add get/settimeofday retargets

Commit:
Adding debugger awarness with Keil MDK

Main thread in Mbed OS is statically allocated and was not available in call
stack of Keil MDK. The RTX5 kernel requires statically allocated thread
information objects that are placed into a specific section to enable RTOS
thread awareness in Keil MDK. This fix is to keep main thread in specific
section of memory.

kegilbert committed:
Add doxygen spellcheck jobs to Travis

Currently tests
    - drivers
    - platform
    - rtos
    - events
    - features/netsocket

paul-szczepanek-arm committed:
fix max payload and hci length values

paul-szczepanek-arm committed:
Docuement active set length and add informative call for querying the limit

paul-szczepanek-arm committed:
add asserts for illegal compinations of adv params

Commit:
Add FLASH support to TARGET_MAX32620C MCU and derivative platform targets

paul-szczepanek-arm committed:
check connectible sizes

paul-szczepanek-arm committed:
check illegal adv params combimnation

Commit:
CMSIS/RTX: Move Idle and Timer thread stack to bss section.

In case of ARM compiler, idle and timer thread stack though assigned
to `.bss.os` section since not zero initialized are part of `data` section.

In this commit, we are moving stacks of idle and timer thread to bss
section and thereby saving ROM space.

mmahadevan108 committed:
Update TEST_APPS to add KW41Z for nanostack tests

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
Enable PSA APIs on Pelion boards

Add PSA label - enabling PSA APIs support on Pelion boards
- K66F
- K66F, NUCLEO_F429ZI
- NUCLEO_F411RE
- UBLOX_EVK_ODIN_W2

Commit:
Add FLASHAIP component

Commit:
Enable entropy injection on PSA targets

enable entropy injection on:
- K64F
- K66F
- NUCLEO_F411RE
- NUCLEO_F429ZI
- UBLOX_EVK_ODIN_W2

ChazJin committed:
Add GD32_E10X as new target

Commit:
Set "TDB_INTERNAL" default storage for Pelion

NUCLEO_F411RE
NUCLEO_F429ZI
UBLOX_EVK_ODIN_W2

Commit:
fix its_reset test

paul-szczepanek-arm committed:
make function name longer

paul-szczepanek-arm committed:
add constructor suggested by Vincent

paul-szczepanek-arm committed:
future proof return size

tz-arm committed:
Initial version for Quectel M26 GSM/GPRS Module.

ChazJin committed:
resolve conflicting file targets.json and add support for GD32_E103VB

Commit:
CMSIS/CORTEX-M: Don't build mbed_tz_context.c in TF-M targets

- TF-M v8m secure-side implements their own TZ context APIs so need to avoid building the Mbed implementation.

kjbracey committed:
LoRAWAN: volatile bool -> atomic_flag

Now we have a proper atomic flag API, use it rather than a volatile
cheat.

paul-szczepanek-arm committed:
simplify for statement

Commit:
Add SHA for "Move Idle and Timer thread stack to bss section" commit

jeromecoutant committed:
FLASHIAP test: skip test if test region overlaps code

Commit:
Update IAR flags for Armv8M devices.

Cortex-M23 / Cortex-M33 CPU settings for baseline and mainline
profile (with optional floating and dsp options) updated.

ciarmcom commented:
@deepikabhavnani, thank you for your changes.
@ARMmbed/mbed-os-tools @ARMmbed/mbed-os-maintainers please review.

Commit:
Sleep manager implements all the API's in platform/mbed_sleep_manager.changes

Not sure why mbed_sleep_manager.c was in hal, moving it to platform

ChazJin committed:
adjust directory structure for device

michalpasztamobica committed:
Gemalto Crinterion UDP test adjustments

Adjusting the tests to match the cellular modems ELS61-E2 and BGS2-W requirements.

costanic committed:
Fix race condition when generating OOB data

The GenericSecurityManager tracks the most recent OOB data generated
by the PAL and the PAL function to generate OOB data is expected to
be asynchronous such that the OOB data is returned via a callback.

There was a race condition on the security manager's oob data variable
because it was cleared (set to all zeros) after calling PAL generate.
The expectation was that the clear operation would occur before the
callback executed, but this is proving to not be the case.  Instead,
the callback is being executed as if it were syncronous with PAL
generate, then PAL generate returns and the oob data is cleared,
thereby losing the generated oob data that was set in the callback.

To fix the issue, clear the oob data variables before calling into
the PAL.

theotherjimmy reviewed (approved):
I'm not comfortable with this coming into a patch release.

deepikabhavnani commented:
> I'm not comfortable with this coming into a patch release.

This is targeted to feature branch, feature branch will be merged in minor release

Commit:
Update the header files

File was moved from hal -> platform folder, hence includes should be
changed accordingly.

theotherjimmy commented:
Thanks, I missed that.

costanic committed:
Fix parameters provided to oob generator function

The function in the Nordic SDK for generating OOB data,
sd_ble_gap_lesc_oob_data_get, requires local LE Secure Connection
P256 Public Keys in {X,Y} format, but was being supplied with
the local secret key.  This caused the generated OOB data to
fail to correspond to the Public Keys, which caused a mismatch
during the OOB pairing phase of the OOB confirmation value by
a remote peer when attempting to verify the OOB data against
the Public Keys, ultimately causing the OOB pairing request to
fail with a Confirm Value Failed (0x04) error.

costanic committed:
Remove own_oob and peer_oob flags from Nordic PAL

the own_oob and peer_oob flags were not being set to 1 even though
an OOB pairing request was in progress, which therefore prevented
OOB data from being passed down to the softdevice during a OOB
pairing operation, thus causing the OOB pairing process to fail.

Commit:
M2351: Update code to support IAR 8.x builds

Commit:
CMSIS/RTX: Update Armv8M IAR 8.x assembly files - add END

Add missing END to the IAR assembly files. In future this commit should be
merged to other assembly file commit, when performing CMSIS update.
https://github.com/ARMmbed/mbed-os/commit/287121ffdc4c9c19f9ce8872e4edd941862daca1

Commit:
Added IAR assembly file update commit to cmsis importer

In future when performing CMSIS update b228cd9db0b7ed79d99f1d9be15d512791f367cc
commit can be merged to 287121ffdc4c9c19f9ce8872e4edd941862daca1 -
CMSIS/RTX: Pre-processor defines used for assembly

bridadan committed:
uARM targets now build in correct ARM_MICRO directory

This is for OS2 testing. Previously, targets that were configured with
the option "default_toolchain": "uARM" would still build to a
TOOLCHAIN_ARM_STD directory. This fixes these targets to build into
TOOLCHAIN_ARM_MICRO.

Commit:
Minor Doxygen fixes for all file system classes

- Remove Doxygen generation on protected functions
- Minor description fixes

NirSonnenschein commented:
The Travis failures are caused by an issue fixed on master a few hours ago, and this change should be propagated here via rebase.  Could you please rebase this PR when it is unparked.

jeromecoutant committed:
STM32: astyle check

deepikabhavnani commented:
> The Travis failures are caused by an issue fixed on master a few hours ago, and this change should be propagated here via rebase. Could you please rebase this PR when it is unparked.

This PR is on top of feature branch, which was rebased  recently. Do i need to rebase feature branch as well?

jeromecoutant committed:
STM32 : typo error in QSPI

Commit:
Update cmsis_importer.json SHAs

- Missing: CMSIS/RTX: Patch to conditionally compile (fb354752eb)
- New: CMSIS/CORTEX-M: Don't build mbed_tz_context.c in TF-M targets (d3f7abdb7c)

jeromecoutant committed:
STM32: replace missing #ifdef DEVICE_xxx

Commit:
[ESP8266] Drop signalling SIGIO artificially if UDP send fails

With TCP it's desirable that SIGIO wakes up the application to check
if there is buffer space space available on the mode. With UDP the
behavior is not acceptable as we don't know if the other endpoint is
there as connection establishment is missing. Hence buffers might
stay full forever.

Commit:
[ESP8266] fixes a debug print

juhaylinen committed:
SDBlockDevice docs update

Commit:
Doxygen update for LoRaRadio class

Added more documentation to the data structures and corrected the
inclusion of this class to the correct class hierarchy.

Commit:
some minor streamlining and editorial changes

Commit:
Update and cleanup BLE.h doxygen

Commit:
Minor Doxygen fixes for BufferedBlockDevice and iBeacon

Commit:
Writer review

Grammatical changes, passive -> active in some places, and future -> present.

Commit:
Some Doxygen fixes in MBRBlockDevice and BlockDevice

Commit:
thread_mle_message_handler: fix build warning

Fix the following build warning found when building with

ARMC6 toolchain for NRF52_DK with mbed cli version 1.8.3

[Warning] thread_mle_message_handler.c@762,0:  #188-D: enumerated type mixed with another type
[Warning] thread_mle_message_handler.c@834,0:  #188-D: enumerated type mixed with another type

Commit:
Review updates

Commit:
Doxygen changes to ProfilingBlockDevice and SlicingBlockDevice.

Commit:
minor editorial tweaks

partition's->partition

Commit:
Merge branch 'yossi_write-a-thon2' of github.com:yossi2le/mbed-os into yossi_write-a-thon2

GuyWi committed:
Update BlockDevice.h

GuyWi committed:
Update iBeacon.h

Technical author review

mmahadevan108 committed:
Supply test config for KW41Z

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
Flush ESP8266 serial input buffers after HW reset

Commit:
Fixed single space that astyle caught

Pharap committed:
Fix bug in templated attach function

The arguments passed to `Callback<void()>`'s constructor were the wrong way round,
thus preventing the `attach` function from being instantiated.
This patch corrects that by switching the order of the arguments.

deepikabhavnani commented:
> The Travis failures are caused by an issue fixed on master a few hours ago, and this change should be propagated here via rebase. Could you please rebase this PR when it is unparked.

This PR is on top of feature branch, which was rebased recently. Do i need to rebase feature branch as well?

Rebased the feature branch and PR

Commit:
Corrected simplified astyle job behavior when base branch is not master

Consolidated efforts from PR #9336.

Previous job's env vars would not be populated correctly if the base branch of a PR was not master.
Corected by pulling remaining respository information to perform comparison between read-only instance of PR and base branch.

mprse committed:
[GigaDevice] Support boot stack size configuration option(GD32_F450ZI)

ChazJin committed:
Support boot stack size configuration option

mprse committed:
mbed_boot_arm_std.c: remove redundant compiler check

Check is not needed since this file is exclusive for ARM compiler.
It causes problems on ARMC6.

mprse committed:
mbed_boot_gcc_arm.c: remove redundant compiler check

0xc0170 committed:
Merge pull request #9346 from cmonr/astyle_in_branch-fix

Travis CI: Corrected astyle job behavior with non-master base branch PRs

simonbutcher committed:
Parameterise repo used in Mbed TLS importer script

This commit makes the repo used to import new Mbed TLS versions into Mbed OS
a parameter, to allow CI scripts (and any other script or users) to specify
specific repos to use for testing.

0xc0170 committed:
Merge pull request #9350 from mprse/adapt_linker_scripts_new_gb_targets

GD32_F450ZI: Support boot stack size configuration option

0xc0170 committed:
Merge pull request #9321 from jeromecoutant/PR_QSPI

STM32 : typo error in QSPI

0xc0170 committed:
Merge pull request #9323 from jeromecoutant/PR_AST

STM32: astyle check

0xc0170 committed:
Merge pull request #9334 from yossi2le/yossi_write-a-thon2

Doxygen fixes in block devices

0xc0170 committed:
Merge pull request #9327 from jeromecoutant/PR_IFDEF

STM32: replace missing #ifdef DEVICE_xxx

0xc0170 committed:
Merge pull request #9305 from davidsaada/david_littlefs_fatfs_doxy_fix

Minor Doxygen fixes for all file system classes

Commit:
[ESP8266] fixes disconnect to check the state from modem

Additionally to internal bookkeeping lets check from the modem
that what is connection state in disconnect. There might be
inconsistencies.

jeromecoutant committed:
NUCLEO_H743ZI : enable TICKLESS

Commit:
Removing the dependency of TDBStore in SystemStorage.h and NVStore co-existence checkup if compile for target TFM

Commit:
Moving KVStore to include sub folder For TFM support

Commit:
Revert "Enable entropy injection on PSA targets"

mbedtls performs selftest on all entropy sources.
Enable injected source without proper entropy injection causes mbedtls
selftest to fail.

Application code can still enable entropy injection by adding the macros
to mbed_app.json/mbed_lib.json

This reverts commit 8c03e3e60d46da71708c4a0e4dacf46a0179d18d.

Commit:
Moving DirectAccessDevicekey.h and DirectAccessDevicekey.cpp to direct_access_devicekey sub folder For TFM support

Commit:
Fix the general block device to support better low memory boards

In addition, fix prints and warnings.

Commit:
Remove unnecessary error prints in FileSystemStore

0xc0170 committed:
Merge pull request #9203 from ChazJin/gd32e10x

Add GD32_E103VB as new target

0xc0170 committed:
Merge pull request #9362 from davidsaada/david_bd_test_low_mem

Fix the general block device for better support of low memory boards

0xc0170 committed:
Merge pull request #9195 from alzix/patch-1

Enable PSA APIs on Pelion boards

0xc0170 committed:
Merge pull request #9330 from hasnainvirk/doxy_loraradio

Doxygen update for LoRaRadio class

0xc0170 committed:
Merge pull request #9355 from sbutcher-arm/mbedtls-importer-repo-param

Parameterise the repo used in the Mbed TLS importer script

0xc0170 committed:
Merge pull request #9295 from deepikabhavnani/iar_asm

Add missing END to Armv8M IAR assembly files

0xc0170 committed:
Merge pull request #9324 from kfnta/avoid_mbed_tz_context

Avoid building mbed_tz_context.c for TF-M targets

0xc0170 committed:
Merge pull request #9341 from SeppoTakalo/flush_esp

Flush ESP8266 serial input buffers after HW reset

0xc0170 committed:
Merge pull request #9332 from jarlamsa/ble_doxygen_cleanup

Update and cleanup BLE.h doxygen

lrusinowicz committed:
Moved nextTimeout variable out of critical section in BLE::callDispatcher.
Initialization of Timeout object contains critical section inside constructor. Initialization inside critical section caused assertion failing under "debug" profile.

Commit:
[ESP8266] adds an error check for data overflow from modem side

0xc0170 committed:
Merge pull request #9329 from juhaylinen/sdblockdevice-doc

SDBlockDevice docs update

tymoteuszblochmobica committed:
Removed cellular driver L3IP stub

fahimalavi committed:
Set target C030-R412M in mbed

0xc0170 committed:
Merge pull request #6622 from NXPmicro/Add_KW41Z_Thread

Add nanostack support for KW41Z

Commit:
Remove TARGET prefix from the TARGET_BYPASS_NVSTORE_CHECK and leave it as BYPASS_NVSTORE_CHECK

0xc0170 committed:
travis: set hypothesis to 3.88 version

Update to v4.x breaks tools, will need to be addressed separately.

michalpasztamobica committed:
TCP/UDP greentea tests refactoring and cleanup

Commit:
Merge pull request #9371 from 0xc0170/test_travis_master

travis: set hypothesis to v3.88

Commit:
add qspif as default storage for LPCXpresso546XX

Commit:
general_block_device test fixed to always erase prior to first program

Commit:
open hal lib source

0xc0170 committed:
Merge pull request #9363 from davidsaada/david_fsst_tr_error_remove

Remove unnecessary error prints in FileSystemStore

0xc0170 committed:
Merge pull request #9357 from jeromecoutant/PR_H7TICKLESS

NUCLEO_H743ZI : enable TICKLESS

0xc0170 committed:
Merge pull request #9356 from VeijoPesonen/bugfix-esp8266_disconnect_return_code

ESP8266 - fixes disconnect to check the state from modem

0xc0170 committed:
Merge pull request #9352 from mprse/armc6_fix

mbed_boot_arm_std.c: remove redundant compiler check

Commit:
MBRBlockDevice: When partitioning, clear the rest of first erase unit

Make sure all the parts of the first erase unit, that are not part of the
partition table are clear.

kapi90 committed:
Fixes typos in SMSC9220 Ethernet driver

Change-Id: I44a12ec1c57a926dadee81df1b84bd531d5cfccb
Signed-off-by: Bence Kaposzta <bence.kaposzta@arm.com>

0xc0170 commented:
Yes please, rebase the feature  branch (another breakages was fixed yesterday)

jeromecoutant committed:
COMPONENT_SD default configuration for ST boards is already defined in TARGET_STM

Commit:
Added pyocd depencency to resolve Travis CI issue

Commit:
Merge pull request #9384 from davidsaada/david_mbr_erase_start

MBRBlockDevice: When partitioning, clear the rest of first erase unit

0xc0170 committed:
Merge pull request #8970 from kegilbert/spell-checker-travisjob

Add doxygen spellcheck job to Travis

0xc0170 committed:
Merge pull request #9245 from kjbracey-arm/lorawan_atomic_flag

LoRAWAN: volatile bool -> atomic_flag

0xc0170 committed:
Merge pull request #9391 from cmonr/bind-pyocd-version

Add pyocd depencency to resolve Travis CI issue

Commit:
Clarify addition to CONFIGURATION.md

Rephrase for clarity.

Commit:
Merge pull request #9374 from maclobdell/LPCXpresso546xx_add_qspif

Add QSPIF as default storage for LPCXpresso546XX

Commit:
Merge pull request #9365 from lrusinowicz/cordio_ble_fix

Fix for asserion failing in BLE::callDispatcher with gcc debug profile.

Commit:
Merge pull request #9361 from yossi2le/yossi_fix_for_tfm

Removing dependency of TDBStore in SystemStorage file and moving some file location - TFM support.

Commit:
Merge pull request #9337 from tymoteuszblochmobica/removecellular

unittest: remove cellular driver L3IP stub

Commit:
Merge pull request #9288 from michalpasztamobica/gemalto_crinterion_test_updates

Gemalto Crinterion UDP test adjustments

Commit:
Merge pull request #9261 from tz-arm/merge_quectel_m26

Initial version for Quectel M26 GSM/GPRS Module.

Commit:
Merge pull request #8497 from kegilbert/gettimeofday_patch

Add get/settimeofday retargets

Commit:
Fixed rtc doxygen spelling errors

Commit:
Added 'retarget' to aspell ignore list

geky committed:
Extended mount to check all metadata-pairs

The most common issue with using littlefs in mbed-os is when users
change from littlefs->FAT->littlefs (or with MBR or similar). When this
corrupts the superblock, littlefs tries to fall back to the backup
superblock. However, at this point in the time the old superblock may be
very out-of-date and pointing to an incorrect filesystem.

There's no complete solution to a malicious modification of the
filesystem (short of checking all metadata+data, a very expensive
operation), but we can at least expand our validation to all of the
metadata for the filesystem. This at least catches the common issues
with changing between different filesystems.

Commit:
DirectAccess DeviceKey added support for default storage type

0xc0170 committed:
Merge pull request #9369 from u-blox/target_C030-R412M

Add target C030-R412M

0xc0170 committed:
Merge pull request #9280 from deepikabhavnani/refactor_a

Sleep manager implements all the API's in platform file

0xc0170 committed:
Merge pull request #9297 from bridadan/fix_os2_tests_uarm

uARM targets now build in correct ARM_MICRO directory

0xc0170 committed:
Merge pull request #9331 from VeijoPesonen/bugfix-esp8266_udp_prevent_would_block

ESP8266: prevent WOULD BLOCK from TX if UDP

0xc0170 committed:
Merge pull request #9096 from pan-/fix-enforce-size-limits

BLE: Enforce advertising data payload limits

0xc0170 committed:
Merge pull request #9405 from cmonr/fix-doxy

Travis: Fix doxy-spellcheck job

0xc0170 committed:
Merge pull request #8887 from deepikabhavnani/uVision_fix

Adding debugger awarness with Keil MDK

Commit:
tools: build_api: remove redundant initialization and fix spelling

Commit:
Merge pull request #9257 from jeromecoutant/PR_FLASHIAP_TIMING

FLASHIAP test: skip test if test region overlaps code

Commit:
Merge pull request #9316 from bentcooke/add_flash_MAX32620FTHR

Add FLASH support to TARGET_MAX32620C and brd targets

Commit:
Merge pull request #9333 from offirko/offir_doxygen

Minor Doxygen fixes for BufferedBlockDevice and iBeacon

Commit:
Merge pull request #9335 from naveenkaje/fix_thred_mle_warning

thread_mle_message_handler: fix build warning

Commit:
Merge pull request #9339 from costanic/fix_oob

Fix Out-Of-Band (OOB) data generation for BLE OOB pairing

Commit:
Merge pull request #9366 from VeijoPesonen/bugfix-esp8266_verify_rcvd_data_len

ESP8266: add an error check for data overflow from modem side

Commit:
Merge pull request #9376 from michalpasztamobica/tcpsocket_greentea

TCP/UDP greentea tests refactoring and cleanup

Commit:
Merge pull request #9380 from kyliuxing/master

UNO_91H: open hal lib source

Commit:
Merge pull request #9403 from ARMmbed/AnotherButler-patch-1

Clarify addition to CONFIGURATION.md

Commit:
Merge pull request #9386 from jeromecoutant/PR_SDJSON

SD default configuration for ST boards is already defined in TARGET_STM

Commit:
Merge pull request #9385 from kapi90/smsc9220_fix

Fixes typos in SMSC9220 Ethernet driver

0xc0170 committed:
Merge pull request #9407 from offirko/offir_direct_access_handle_default_config

DirectAccess DeviceKey: add support for default storage type

0xc0170 committed:
Merge pull request #9406 from geky/littlefs-validate-all-dirs

Extended mount to check all metadata-pairs

0xc0170 committed:
Merge pull request #9344 from Pharap/fix-attach-function

Fix bug in templated attach function

0xc0170 committed:
Merge pull request #9377 from offirko/offir_general_block_device_test

test: general_block_device fix to always erase prior the first program

0xc0170 committed:
Merge pull request #9418 from naveenkaje/build_api_spell_fix

tools: build_api: remove redundant initialization and fix spelling

deepikabhavnani commented:
Closing this and creating new PR - one on mater and other on  feature branch
